lib/dfu.js

```diff
@@ -295,5 +299,54 @@ var that = module.exports = {
 		);
 	},
 
+	_udevRulesDir: '/etc/udev/rules.d/',
+
+	_missingDevicePermissions: function (stderr) {
+		return stderr && stderr.indexOf("Cannot open DFU device") >= 0;
```

Comment:
Ok. Can you add a check for the case when the rules are installed but there is still a permissions issue? We shouldn't prompt to install the rules in this case.